### ISSUE TITLE
🚀 Feature: Add a new extension point to enable custom transport in the notifications-backend-module-email module.

### ISSUE THREAD

**Issue by franvivas** [backstage/backstage#28119]
### 🔖 Feature description

The notifications-backend-module-email plugin supports four types of transport (smtp, ses, sendmail, stream) for sending emails. 
These transport options can be configured via app-config.yaml. However, the transport mechanism is not extensible, meaning it is not possible to use this plugin with other types of transport. 
It would be beneficial for the plugin to provide an extension point that allows users to inject custom transport implementations.

### 🎤 Context

In my company, we use @microsoft/microsoft-graph-client to send emails. The other features of the notifications-backend-module-email plugin, such as retrieving users and broadcasting, are perfect for our needs. However, we cannot use the plugin because this transport option is not implemented.

I believe there are others with similar requirements.

### ✌️ Possible Implementation

I would like to have an extension point that it would be something like this: 

export interface NotificationEmailSender {
  sendMail(options: Mail.Options)): Promise<void>;
}

/**
 * @public
 */
export interface NotificationsEmailTemplateExtensionPoint {
  setTemplateRenderer(renderer: NotificationTemplateRenderer): void;
  setEmailSender(NotificationEmailSender): void;
}

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by freben** [comment]
Pinging @backstage/notifications-maintainers on this one!

In general, we try to avoid making module extension points that people will commonly want to make published packages for. The reason for this is that it's a bit semantically unclear how to package up such a thing. So our hunch is, that the best outcome here may be to contribute more builtin transports in OSS, than making them pluggable.

**Comment by drodil** [comment]
Agreed with above. Would make either sense to create a new backend module for notifications for graph email only (basically copy from the current email module but with different send implementation) OR extend the email module to support this. Extension point on top of another extension point doesn't sound very nice.

**Comment by franvivas** [comment]
Thanks, guys. In the end, we’re doing it just as @drodil mentioned. Once we have the implementation done, we’ll let you know in case we make a contribution. Many thx.
